consensus.go

```diff
@@ -0,0 +1,235 @@
+package dcrlibwallet
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"decred.org/dcrwallet/v2/errors"
+	w "decred.org/dcrwallet/v2/wallet"
+	"github.com/decred/dcrd/chaincfg/chainhash"
+	"github.com/decred/dcrd/chaincfg/v3"
+)
+
+// GetVoteChoices returns configured vote preferences for each agenda of the
+// latest supported stake version for the wallet with the specified walletID.
+func (mw *MultiWallet) GetVoteChoices(hash string, walletID int) (*GetVoteChoicesResult, error) {
+	wallet := mw.WalletWithID(walletID)
+	if wallet == nil {
+		return nil, errors.New(ErrNotExist)
+	}
+	wal := wallet.Internal()
+
+	var ticketHash *chainhash.Hash
+	if hash != "" {
+		hash, err := chainhash.NewHashFromStr(hash)
+		if err != nil {
+			return nil, fmt.Errorf("inavlid hash: %w", err)
+		}
+		ticketHash = hash
+	}
+
+	version, agendas := getAllAgendas(wal.ChainParams())
+	resp := &GetVoteChoicesResult{
+		Version: version,
+		Choices: make([]AgendaVoteChoice, len(agendas)),
+	}
+
+	ctx := wallet.shutdownContext()
+	choices, _, err := wal.AgendaChoices(ctx, ticketHash)
+	if err != nil {
+		return nil, err
+	}
+
+	for i := range choices {
+		agenda := mw.agenda(agendas, choices[i].AgendaID)
+		resp.Choices[i] = AgendaVoteChoice{
+			AgendaID:          choices[i].AgendaID,
+			AgendaDescription: agenda.Vote.Description,
+			ChoiceID:          choices[i].ChoiceID,
+			ChoiceDescription: "", // Set below
```

Comment:
Since this will be set below in same loop, there is no need setting it to `""` as its default values is `""`

Reply:
That's the way the code is done upstream, and i believe the reason for that is for better code readability, because at first glace if that variable isn't there and commented, the for loop might not be so obvious